### DISCUSSION TITLE
source config dynamic loading

### DISCUSSION THREAD

**Discussion by tuoeg** [lf-edge/ekuiper#1404] [Q&A]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
the source config in /etc/source/xx.yaml.if I modify xx.yaml,whether the source connection will update when i don't restart ekuiper,or the 
source config will update when i create stream or rule,please help me,thanks

**Comment by ngjaying** [comment]
The new configuration will be loaded when rule starting. So your running rule which is using the source will adopt the new configuration after restart. The new created rule will adopt the new configuration.

**Comment by tuoeg** [comment]
> 
whether i also need't call ReadSourceMetaFile() function,when i update the source config file
![image](https://user-images.githubusercontent.com/49616374/190327001-fea36322-e7c3-433e-8ae3-9316a892a281.png)

**Comment by ngjaying** [comment]
No, this API is irrelevant. It is just to provide cofiguration CRUD by REST API.

**Comment by tuoeg** [comment]
> No, this API is irrelevant. It is just to provide cofiguration CRUD by REST API.

ok,thans you!
